nni/compression/pytorch/utils/counter.py

```diff
@@ -1,18 +1,231 @@
+        # https://en.wikipedia.org/wiki/Bicubic_interpolation
+        # Product matrix [4x4] x [4x4] x [4x4]
+        ops_solve_A = 224  # 128 muls + 96 adds
+        ops_solve_p = 35  # 16 muls + 12 adds + 4 muls + 3 adds
+        total_ops = y.nelement() * (ops_solve_A + ops_solve_p)
+    elif m.mode == "trilinear":
+        # https://en.wikipedia.org/wiki/Trilinear_interpolation
+        # can viewed as 2 bilinear + 1 linear
+        total_ops = y.nelement() * (13 * 2 + 5)
+    else:
+        return None
+
+    results = {
+        "flops": total_ops,
+        "params": 0,
+        "weight_size": 0,
+        "input_size": tuple(x[0].size()),
+        "output_size": tuple(y.size()),
+        "module_type": type(m).__name__
+    }
+
+    return results
+
+DEFAULT_OPS = {
+    nn.Conv1d: count_convNd,
+    nn.Conv2d: count_convNd,
+    nn.Conv3d: count_convNd,
+    nn.Linear: count_linear
+}
+
+OTHER_OPS = {
+    nn.ConvTranspose1d: count_convNd,
+    nn.ConvTranspose2d: count_convNd,
+    nn.ConvTranspose3d: count_convNd,
+    nn.BatchNorm1d: count_bn,
+    nn.BatchNorm2d: count_bn,
+    nn.BatchNorm3d: count_bn,
+    nn.LeakyReLU: count_relu,
+    nn.AvgPool1d: count_avgpool,
+    nn.AvgPool2d: count_avgpool,
+    nn.AvgPool3d: count_avgpool,
+    nn.AdaptiveAvgPool1d: count_adap_avgpool,
+    nn.AdaptiveAvgPool2d: count_adap_avgpool,
+    nn.AdaptiveAvgPool3d: count_adap_avgpool,
+    nn.Upsample: count_upsample,
+    nn.UpsamplingBilinear2d: count_upsample,
+    nn.UpsamplingNearest2d: count_upsample
+}
+
+mode_cls = {
+    'default': DEFAULT_OPS,
+    'full': {k: v for d in [DEFAULT_OPS, OTHER_OPS] for k, v in d.items()}
+}
+
+
+def format_results(modules):
+    table = PrettyTable()
+    has_multi_use = True if len(list(filter(lambda x: len(x) > 1, [module['flops'] for module in modules.values()]))) > 0 else False
+
+    headers = [
+        'Index',
+        'Name',
+        'Module',
+        'Weight Size',
+        'FLOPs',
+        '#Params',
+    ]
+    if has_multi_use:
+        headers.append('#Calls')
+
+    table.field_names = headers
+    total_ops, total_params = 0, 0
+    for i, (name, module) in enumerate(modules.items()):
+        if len(module['flops']) == 0: continue
+        row_values = [
+            i,
+            name,  # Name
+            module['module_type'][0],
+            str(module['weight_size'][0]),
+            sum(module['flops']),
+            module['params'][0],
+        ]
+        if has_multi_use:
+            row_values.append(len(module['flops']))
+
+        total_ops += row_values[4]
+        total_params += row_values[5]
+        table.add_row(row_values)
+    return table
+
+
+def count_module_flops_params(m, x, y):
+    m_type = type(m)
+    if m_type not in register_ops.keys():
+        return
+
+    fn = register_ops[m_type]
+    module_result = fn(m, x, y)
+    if module_result is None:
+        return
+
+    for k, v in module_result.items():
+        if k not in results[m._name]:
```

Comment:
Maybe register results as a local variable in `count_flops_params` is a good idea. It can still be accessible to `count_module_flops_params` via `functools.partial`.